### ISSUE TITLE
Allowing cursor positioning on UI.Input

### ISSUE THREAD

**Issue by Jumbli** [StereoKit/StereoKit#434]
## Describe the feature
If you make an early typing mistake on an input field, you have to delete all the characters to get back to the mistake and then rekey the rest of the word. This is very frustrating when you make as many typing mistakes as I do :)

Can you please support positioning the cursor where you touch the input field and also moving the cursor using the arrow keys. You already have a Text.Size function so it should be easy to run through the input string to calculate the various offsets for each character and set the proper cursor position when the field is touched.

You could have an additional "ref cursorPosition" input to UI.Text so you don't need to maintain the cursor position internally.

Thanks.

**Comment by maluoi** [comment]
Yeah, this is probably a pretty important one that I've put off a little too long 😬 I'll put this on the list for this version, and try to look into this soon!

**Comment by maluoi** [comment]
Some initial progress towards cursor/carat editing! Still needs a few visual fixes, probably a different method for supporting overflow, and finger based carat positioning. Would also like to support word skipping, like what happens when you hold down Ctrl and arrow around.

![StereoKitCaratInput](https://user-images.githubusercontent.com/8484884/179451238-292b8569-5bc0-451d-b750-2239b3dd2378.gif)

**Comment by Jumbli** [comment]
Looking good. If you can get the finger based carat positioning in, that would be great. I think people will have an assumption that it will work like a mobile phone input.
My preference for overflow would be for the text size to remain consistent (or greater than a set minimum size) and the text box to expand as required.
